trieste/acquisition/_optimizer.py

```diff
@@ -62,11 +62,16 @@ def _box(space: Box, target_func: TensorMapping) -> QueryPoints:
     initial_point = optimize(trial_search_space, target_func)
 
     bijector = tfp.bijectors.Sigmoid(low=space.lower, high=space.upper)
-    variable = tf.Variable(bijector.inverse(initial_point))
 
-    def _objective() -> tf.Tensor:
-        return -target_func(bijector.forward(variable))
+    def _objective(x: tf.Variable) -> tf.Tensor:
+        return -target_func(bijector.forward(x))[..., 0]
 
-    gpflow.optimizers.Scipy().minimize(_objective, (variable,))
+    lbfgs_result = tfp.optimizer.lbfgs_minimize(
+        lambda x: tfp.math.value_and_gradient(_objective, x),
+        tf.Variable(bijector.inverse(initial_point))
+    )
+
+    if not lbfgs_result.converged:
+        raise RuntimeError("L-BFGS optimization failed to converge.")
```

Comment:
should we raise if it didn't converge?

Reply:
apparently not, because points can still be useful even if convergence wasn't successful